satellite/satellitedb/storagenodeaccounting.go

```diff
@@ -200,21 +200,49 @@ func (db *StoragenodeAccounting) QueryPaymentInfo(ctx context.Context, start tim
 func (db *StoragenodeAccounting) QueryStorageNodeUsage(ctx context.Context, nodeID storj.NodeID, start time.Time, end time.Time) (_ []accounting.StorageNodeUsage, err error) {
 	defer mon.Task()(&ctx)(&err)
 
-	query := `SELECT at_rest_total, start_time, 
-		LAG(at_rest_total) OVER win AS prev_at_rest, 
-		LAG(start_time) OVER win AS prev_start_time
-		FROM accounting_rollups
-		WHERE id IN (
-			SELECT MAX(id)
-			FROM accounting_rollups
-			WHERE node_id = ?
-			AND ? <= start_time AND start_time <= ?
-			GROUP BY start_time
-			ORDER BY start_time ASC
-		)
-		WINDOW win AS (ORDER BY start_time)`
+	lastRollup, err := db.db.Find_AccountingTimestamps_Value_By_Name(ctx, dbx.AccountingTimestamps_Name(accounting.LastRollup))
+	if err != nil {
+		return nil, Error.Wrap(err)
+	}
+	if lastRollup == nil {
+		return nil, nil
+	}
+
+	start, end = start.UTC(), end.UTC()
+
+	query := `WITH r AS (
+				SELECT at_rest_total, start_time
+				FROM accounting_rollups
+				WHERE node_id = ?
+				AND ? <= start_time AND start_time <= ?
+			)
+			SELECT at_rest_total, start_time from r
+				UNION
+					SELECT SUM(data_total) AS at_rest_total, DATE(interval_end_time) AS start_time
+					FROM storagenode_storage_tallies
+					WHERE node_id = ?
+					AND NOT EXISTS (SELECT 1 FROM r WHERE DATE(r.start_time) = DATE(interval_end_time))
+					AND (SELECT value FROM accounting_timestamps WHERE name = ?) < interval_end_time AND interval_end_time <= ?
+					GROUP BY start_time
+			ORDER BY start_time`
+
+	//query := `SELECT at_rest_total, start_time
```

Comment:
Please make sure to delete this before merging

Reply:
Yes, one of this query will be deleted as we need to leave one with best performance